hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/HCFSMountTableConfigLoader.java

```diff
@@ -0,0 +1,111 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An implementation for Apache Hadoop compatible file system based mount-table
+ * file loading.
+ */
+public class HCFSMountTableConfigLoader implements MountTableConfigLoader {
+  private static final String REGEX_DOT = "[.]";
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(HCFSMountTableConfigLoader.class);
+  private Path mountTablePath = null;
+
+  /**
+   * Loads the mount-table configuration from hadoop compatible file system and
+   * add the configuration items to given configuration. Mount-table
+   * configuration format should be suffixed with version number. Format:
+   * mount-table.<versionNumber>.xml Example: mount-table.1.xml When user wants
+   * to update mount table, the expectation is to upload new configuration file
+   * with incremented version number. This API loads the highest version number
+   * file. We can also configure single file path directly.
+   *
+   * @param mountTableConfigPath : A directory path where mount-table files
+   *          stored or a mount-table file path. We recommend to configure
+   *          directory with the mount-table version files.
+   * @param conf : to add the mount table as resource.
+   */
+  @Override
+  public void load(String mountTableConfigPath, Configuration conf)
+      throws IOException {
+    this.mountTablePath = new Path(mountTableConfigPath);
+    String scheme = mountTablePath.toUri().getScheme();
+    ViewFileSystem.FsGetter fsGetter =
+        new ViewFileSystemOverloadScheme.ChildFsGetter(scheme);
+    FileSystem fs = fsGetter.getNewInstance(mountTablePath.toUri(), conf);
+
+    RemoteIterator<LocatedFileStatus> listFiles =
+        fs.listFiles(mountTablePath, false);
+    LocatedFileStatus lfs = null;
+    int higheirVersion = -1;
+    while (listFiles.hasNext()) {
+      LocatedFileStatus curLfs = listFiles.next();
+      String cur = curLfs.getPath().getName();
+      String[] nameParts = cur.split(REGEX_DOT);
+      if (nameParts.length < 2) {
+        logInvalidFileNameFormat(cur);
+        continue; // invalid file name
+      }
+      int curVersion = higheirVersion;
+      try {
+        curVersion = Integer.parseInt(nameParts[nameParts.length - 2]);
+      } catch (NumberFormatException nfe) {
+        logInvalidFileNameFormat(cur);
+        continue;
+      }
+
+      if (curVersion > higheirVersion) {
+        higheirVersion = curVersion;
+        lfs = curLfs;
+      }
+    }
+
+    if (lfs == null) {
+      // No valid mount table file found.
+      // TODO: Should we fail? Currently viewfs init will fail if no mount
+      // links anyway.
+      LOGGER.warn("No valid mount-table file exist at: " + mountTableConfigPath
+          + ". At least one mount-table file should present with the name "
+          + "format: mount-table.<versionNumber>.xml");
+      return;
+    }
+    // Latest version file.
+    Path letestVersionMountTable = lfs.getPath();
```

Comment:
can we log a debug message to print the mount table file name we are about to load?

Reply:
Good point. I will add a log here.